diff --git a/oqsprov/oqs_kmgmt.c b/oqsprov/oqs_kmgmt.c
index 3814ca6..1de0178 100644
--- a/oqsprov/oqs_kmgmt.c
+++ b/oqsprov/oqs_kmgmt.c
@@ -1523,6 +1523,10 @@ MAKE_KEM_KEYMGMT_FUNCTIONS(frodo1344shake, OQS_KEM_alg_frodokem_1344_shake, 256)
 
 MAKE_KEM_ECP_KEYMGMT_FUNCTIONS(p521_frodo1344shake,
                                OQS_KEM_alg_frodokem_1344_shake, 256)
+MAKE_KEM_KEYMGMT_FUNCTIONS(kyber1024, OQS_KEM_alg_kyber_1024, 256)
+
+MAKE_KEM_ECP_KEYMGMT_FUNCTIONS(p521_kyber1024, OQS_KEM_alg_kyber_1024, 256)
+MAKE_KEM_ECP_KEYMGMT_FUNCTIONS(secp521_r1_kyber1024, OQS_KEM_alg_kyber_1024, 256)
 MAKE_KEM_KEYMGMT_FUNCTIONS(mlkem512, OQS_KEM_alg_ml_kem_512, 128)
 
 MAKE_KEM_ECP_KEYMGMT_FUNCTIONS(p256_mlkem512, OQS_KEM_alg_ml_kem_512, 128)
