diff --git a/oqsprov/oqs_kmgmt.c b/oqsprov/oqs_kmgmt.c
index fb7323c..1ea008e 100644
--- a/oqsprov/oqs_kmgmt.c
+++ b/oqsprov/oqs_kmgmt.c
@@ -164,7 +164,7 @@ static int oqsx_export(void *keydata, int selection, OSSL_CALLBACK *param_cb,
         goto err;
 
     ret = param_cb(params, cbarg);
-    OSSL_PARAM_BLD_free_params(params);
+    OSSL_PARAM_free(params);
 err:
     OSSL_PARAM_BLD_free(tmpl);
     return ret;
@@ -396,154 +396,154 @@ static int oqsx_gen_set_params(void *genctx, const OSSL_PARAM params[])
 ///// OQS_TEMPLATE_FRAGMENT_KEYMGMT_CONSTRUCTORS_START
 static void *oqs_sig_default_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_default, "oqs_sig_default", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DEFAULT", "oqs_sig_default", 0, NULL);
 }
 
 static void *oqs_sig_default_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_default, 0);
+    return oqsx_gen_init(provctx, selection, "DEFAULT", 0);
 }
 
 static void *dilithium2_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_2, "dilithium2", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_2", "dilithium2", 0, NULL);
 }
 
 static void *dilithium2_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_2, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_2", 0);
 }
 static void *dilithium3_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_3, "dilithium3", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_3", "dilithium3", 0, NULL);
 }
 
 static void *dilithium3_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_3, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_3", 0);
 }
 static void *dilithium5_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_5, "dilithium5", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_5", "dilithium5", 0, NULL);
 }
 
 static void *dilithium5_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_5, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_5", 0);
 }
 static void *dilithium2_aes_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_2_aes, "dilithium2_aes", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_2_AES", "dilithium2_aes", 0, NULL);
 }
 
 static void *dilithium2_aes_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_2_aes, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_2_AES", 0);
 }
 static void *dilithium3_aes_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_3_aes, "dilithium3_aes", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_3_AES", "dilithium3_aes", 0, NULL);
 }
 
 static void *dilithium3_aes_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_3_aes, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_3_AES", 0);
 }
 static void *dilithium5_aes_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_dilithium_5_aes, "dilithium5_aes", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "DILITHIUM_5_AES", "dilithium5_aes", 0, NULL);
 }
 
 static void *dilithium5_aes_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_dilithium_5_aes, 0);
+    return oqsx_gen_init(provctx, selection, "DILITHIUM_5_AES", 0);
 }
 
 static void *falcon512_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_falcon_512, "falcon512", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "FALCON_512", "falcon512", 0, NULL);
 }
 
 static void *falcon512_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_falcon_512, 0);
+    return oqsx_gen_init(provctx, selection, "FALCON_512", 0);
 }
 static void *falcon1024_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_falcon_1024, "falcon1024", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "FALCON_1024", "falcon1024", 0, NULL);
 }
 
 static void *falcon1024_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_falcon_1024, 0);
+    return oqsx_gen_init(provctx, selection, "FALCON_1024", 0);
 }
 
 static void *picnicl1full_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_picnic_L1_full, "picnicl1full", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "PICNIC_L1_FULL", "picnicl1full", 0, NULL);
 }
 
 static void *picnicl1full_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_picnic_L1_full, 0);
+    return oqsx_gen_init(provctx, selection, "PICNIC_L1_FULL", 0);
 }
 static void *picnic3l1_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_picnic3_L1, "picnic3l1", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "PICNIC3_L1", "picnic3l1", 0, NULL);
 }
 
 static void *picnic3l1_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_picnic3_L1, 0);
+    return oqsx_gen_init(provctx, selection, "PICNIC3_L1", 0);
 }
 
 static void *rainbowIclassic_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_rainbow_I_classic, "rainbowIclassic", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "RAINBOW_I_CLASSIC", "rainbowIclassic", 0, NULL);
 }
 
 static void *rainbowIclassic_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_rainbow_I_classic, 0);
+    return oqsx_gen_init(provctx, selection, "RAINBOW_I_CLASSIC", 0);
 }
 static void *rainbowVclassic_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_rainbow_V_classic, "rainbowVclassic", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "RAINBOW_V_CLASSIC", "rainbowVclassic", 0, NULL);
 }
 
 static void *rainbowVclassic_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_rainbow_V_classic, 0);
+    return oqsx_gen_init(provctx, selection, "RAINBOW_V_CLASSIC", 0);
 }
 
 static void *sphincsharaka128frobust_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_sphincs_haraka_128f_robust, "sphincsharaka128frobust", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SPHINCS_HARAKA_128F_ROBUST", "sphincsharaka128frobust", 0, NULL);
 }
 
 static void *sphincsharaka128frobust_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_sphincs_haraka_128f_robust, 0);
+    return oqsx_gen_init(provctx, selection, "SPHINCS_HARAKA_128F_ROBUST", 0);
 }
 
 static void *sphincssha256128frobust_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_sphincs_sha256_128f_robust, "sphincssha256128frobust", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SPHINCS_SHA256_128F_ROBUST", "sphincssha256128frobust", 0, NULL);
 }
 
 static void *sphincssha256128frobust_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_sphincs_sha256_128f_robust, 0);
+    return oqsx_gen_init(provctx, selection, "SPHINCS_SHA256_128F_ROBUST", 0);
 }
 
 static void *sphincsshake256128frobust_new_key(void *provctx)
 {
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_SIG_alg_sphincs_shake256_128f_robust, "sphincsshake256128frobust", 0, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SPHINCS_SHAKE256_128F_ROBUST", "sphincsshake256128frobust", 0, NULL);
 }
 
 static void *sphincsshake256128frobust_gen_init(void *provctx, int selection)
 {
-    return oqsx_gen_init(provctx, selection, OQS_SIG_alg_sphincs_shake256_128f_robust, 0);
+    return oqsx_gen_init(provctx, selection, "SPHINCS_SHAKE256_128F_ROBUST", 0);
 }
 
 
@@ -610,22 +610,22 @@ static void *frodo1344shake_gen_init(void *provctx, int selection)
 
 static void *bike1l1cpa_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_bike1_l1_cpa, "bike1l1cpa", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "BIKE1_L1_CPA", "bike1l1cpa", 1, NULL);
 }
 
 static void *bike1l1cpa_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_bike1_l1_cpa, 1);
+    return oqsx_gen_init(provctx, selection, "BIKE1_L1_CPA", 1);
 } 
 
 static void *bike1l3cpa_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_bike1_l3_cpa, "bike1l3cpa", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "BIKE1_L3_CPA", "bike1l3cpa", 1, NULL);
 }
 
 static void *bike1l3cpa_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_bike1_l3_cpa, 1);
+    return oqsx_gen_init(provctx, selection, "BIKE1_L3_CPA", 1);
 } 
 
 static void *kyber512_new_key(void *provctx)
@@ -700,252 +700,252 @@ static void *ntru_hrss701_gen_init(void *provctx, int selection)
 
 static void *lightsaber_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_saber_lightsaber, "lightsaber", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SABER_LIGHTSABER", "lightsaber", 1, NULL);
 }
 
 static void *lightsaber_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_saber_lightsaber, 1);
+    return oqsx_gen_init(provctx, selection, "SABER_LIGHTSABER", 1);
 } 
 
 static void *saber_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_saber_saber, "saber", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SABER_SABER", "saber", 1, NULL);
 }
 
 static void *saber_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_saber_saber, 1);
+    return oqsx_gen_init(provctx, selection, "SABER_SABER", 1);
 } 
 
 static void *firesaber_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_saber_firesaber, "firesaber", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SABER_FIRESABER", "firesaber", 1, NULL);
 }
 
 static void *firesaber_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_saber_firesaber, 1);
+    return oqsx_gen_init(provctx, selection, "SABER_FIRESABER", 1);
 } 
 
 static void *sidhp434_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sidh_p434, "sidhp434", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIDH_P434", "sidhp434", 1, NULL);
 }
 
 static void *sidhp434_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sidh_p434, 1);
+    return oqsx_gen_init(provctx, selection, "SIDH_P434", 1);
 } 
 
 static void *sidhp503_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sidh_p503, "sidhp503", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIDH_P503", "sidhp503", 1, NULL);
 }
 
 static void *sidhp503_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sidh_p503, 1);
+    return oqsx_gen_init(provctx, selection, "SIDH_P503", 1);
 } 
 
 static void *sidhp610_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sidh_p610, "sidhp610", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIDH_P610", "sidhp610", 1, NULL);
 }
 
 static void *sidhp610_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sidh_p610, 1);
+    return oqsx_gen_init(provctx, selection, "SIDH_P610", 1);
 } 
 
 static void *sidhp751_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sidh_p751, "sidhp751", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIDH_P751", "sidhp751", 1, NULL);
 }
 
 static void *sidhp751_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sidh_p751, 1);
+    return oqsx_gen_init(provctx, selection, "SIDH_P751", 1);
 } 
 
 static void *sikep434_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sike_p434, "sikep434", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIKE_P434", "sikep434", 1, NULL);
 }
 
 static void *sikep434_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sike_p434, 1);
+    return oqsx_gen_init(provctx, selection, "SIKE_P434", 1);
 } 
 
 static void *sikep503_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sike_p503, "sikep503", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIKE_P503", "sikep503", 1, NULL);
 }
 
 static void *sikep503_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sike_p503, 1);
+    return oqsx_gen_init(provctx, selection, "SIKE_P503", 1);
 } 
 
 static void *sikep610_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sike_p610, "sikep610", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIKE_P610", "sikep610", 1, NULL);
 }
 
 static void *sikep610_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sike_p610, 1);
+    return oqsx_gen_init(provctx, selection, "SIKE_P610", 1);
 } 
 
 static void *sikep751_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_sike_p751, "sikep751", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "SIKE_P751", "sikep751", 1, NULL);
 }
 
 static void *sikep751_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_sike_p751, 1);
+    return oqsx_gen_init(provctx, selection, "SIKE_P751", 1);
 } 
 
 static void *bike1l1fo_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_bike1_l1_fo, "bike1l1fo", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "BIKE1_L1_FO", "bike1l1fo", 1, NULL);
 }
 
 static void *bike1l1fo_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_bike1_l1_fo, 1);
+    return oqsx_gen_init(provctx, selection, "BIKE1_L1_FO", 1);
 } 
 
 static void *bike1l3fo_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_bike1_l3_fo, "bike1l3fo", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "BIKE1_L3_FO", "bike1l3fo", 1, NULL);
 }
 
 static void *bike1l3fo_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_bike1_l3_fo, 1);
+    return oqsx_gen_init(provctx, selection, "BIKE1_L3_FO", 1);
 } 
 
 static void *kyber90s512_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_kyber_512_90s, "kyber90s512", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "KYBER_512_90S", "kyber90s512", 1, NULL);
 }
 
 static void *kyber90s512_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_kyber_512_90s, 1);
+    return oqsx_gen_init(provctx, selection, "KYBER_512_90S", 1);
 } 
 
 static void *kyber90s768_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_kyber_768_90s, "kyber90s768", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "KYBER_768_90S", "kyber90s768", 1, NULL);
 }
 
 static void *kyber90s768_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_kyber_768_90s, 1);
+    return oqsx_gen_init(provctx, selection, "KYBER_768_90S", 1);
 } 
 
 static void *kyber90s1024_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_kyber_1024_90s, "kyber90s1024", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "KYBER_1024_90S", "kyber90s1024", 1, NULL);
 }
 
 static void *kyber90s1024_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_kyber_1024_90s, 1);
+    return oqsx_gen_init(provctx, selection, "KYBER_1024_90S", 1);
 } 
 
 static void *hqc128_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_hqc_128, "hqc128", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "HQC_128", "hqc128", 1, NULL);
 }
 
 static void *hqc128_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_hqc_128, 1);
+    return oqsx_gen_init(provctx, selection, "HQC_128", 1);
 } 
 
 static void *hqc192_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_hqc_192, "hqc192", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "HQC_192", "hqc192", 1, NULL);
 }
 
 static void *hqc192_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_hqc_192, 1);
+    return oqsx_gen_init(provctx, selection, "HQC_192", 1);
 } 
 
 static void *hqc256_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_hqc_256, "hqc256", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "HQC_256", "hqc256", 1, NULL);
 }
 
 static void *hqc256_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_hqc_256, 1);
+    return oqsx_gen_init(provctx, selection, "HQC_256", 1);
 } 
 
 static void *ntrulpr653_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_ntrulpr653, "ntrulpr653", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_NTRULPR653", "ntrulpr653", 1, NULL);
 }
 
 static void *ntrulpr653_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_ntrulpr653, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_NTRULPR653", 1);
 } 
 
 static void *ntrulpr761_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_ntrulpr761, "ntrulpr761", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_NTRULPR761", "ntrulpr761", 1, NULL);
 }
 
 static void *ntrulpr761_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_ntrulpr761, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_NTRULPR761", 1);
 } 
 
 static void *ntrulpr857_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_ntrulpr857, "ntrulpr857", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_NTRULPR857", "ntrulpr857", 1, NULL);
 }
 
 static void *ntrulpr857_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_ntrulpr857, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_NTRULPR857", 1);
 } 
 
 static void *sntrup653_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_sntrup653, "sntrup653", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_SNTRUP653", "sntrup653", 1, NULL);
 }
 
 static void *sntrup653_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_sntrup653, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_SNTRUP653", 1);
 } 
 
 static void *sntrup761_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_sntrup761, "sntrup761", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_SNTRUP761", "sntrup761", 1, NULL);
 }
 
 static void *sntrup761_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_sntrup761, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_SNTRUP761", 1);
 } 
 
 static void *sntrup857_new_key(void *provctx)
 { 
-    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), OQS_KEM_alg_ntruprime_sntrup857, "sntrup857", 1, NULL);
+    return oqsx_key_new(PROV_OQS_LIBCTX_OF(provctx), "NTRUPRIME_SNTRUP857", "sntrup857", 1, NULL);
 }
 
 static void *sntrup857_gen_init(void *provctx, int selection)
 { 
-    return oqsx_gen_init(provctx, selection, OQS_KEM_alg_ntruprime_sntrup857, 1);
+    return oqsx_gen_init(provctx, selection, "NTRUPRIME_SNTRUP857", 1);
 } 
 
 ///// OQS_TEMPLATE_FRAGMENT_KEYMGMT_CONSTRUCTORS_END
