diff --git a/oqsprov/oqs_kem.c b/oqsprov/oqs_kem.c
index dbaa7a5..184d6c7 100644
--- a/oqsprov/oqs_kem.c
+++ b/oqsprov/oqs_kem.c
@@ -28,9 +28,9 @@
 
 
 static OSSL_FUNC_kem_newctx_fn oqs_kem_newctx;
-static OSSL_FUNC_kem_encapsulate_init_fn oqs_kem_encaps_init;
+
 static OSSL_FUNC_kem_encapsulate_fn oqs_kem_encaps;
-static OSSL_FUNC_kem_decapsulate_init_fn oqs_kem_decaps_init;
+
 static OSSL_FUNC_kem_decapsulate_fn oqs_kem_decaps;
 static OSSL_FUNC_kem_freectx_fn oqs_kem_freectx;
 
