diff --git a/oqsprov/oqs_sig.c b/oqsprov/oqs_sig.c
index 76f79ca..700362b 100644
--- a/oqsprov/oqs_sig.c
+++ b/oqsprov/oqs_sig.c
@@ -49,76 +49,76 @@
 #endif // NDEBUG
 
 static OSSL_FUNC_signature_newctx_fn oqs_sig_newctx;
-static OSSL_FUNC_signature_sign_init_fn oqs_sig_sign_init;
-static OSSL_FUNC_signature_verify_init_fn oqs_sig_verify_init;
+
+
 static OSSL_FUNC_signature_sign_fn oqs_sig_sign;
 static OSSL_FUNC_signature_verify_fn oqs_sig_verify;
-static OSSL_FUNC_signature_digest_sign_init_fn oqs_sig_digest_sign_init;
+
 static OSSL_FUNC_signature_digest_sign_update_fn oqs_sig_digest_signverify_update;
 static OSSL_FUNC_signature_digest_sign_final_fn oqs_sig_digest_sign_final;
-static OSSL_FUNC_signature_digest_verify_init_fn oqs_sig_digest_verify_init;
+
 static OSSL_FUNC_signature_digest_verify_update_fn oqs_sig_digest_signverify_update;
 static OSSL_FUNC_signature_digest_verify_final_fn oqs_sig_digest_verify_final;
 static OSSL_FUNC_signature_freectx_fn oqs_sig_freectx;
 static OSSL_FUNC_signature_dupctx_fn oqs_sig_dupctx;
 static OSSL_FUNC_signature_get_ctx_params_fn oqs_sig_get_ctx_params;
-static OSSL_FUNC_signature_gettable_ctx_params_fn oqs_sig_gettable_ctx_params;
+
 static OSSL_FUNC_signature_set_ctx_params_fn oqs_sig_set_ctx_params;
-static OSSL_FUNC_signature_settable_ctx_params_fn oqs_sig_settable_ctx_params;
+
 static OSSL_FUNC_signature_get_ctx_md_params_fn oqs_sig_get_ctx_md_params;
-static OSSL_FUNC_signature_gettable_ctx_md_params_fn oqs_sig_gettable_ctx_md_params;
+
 static OSSL_FUNC_signature_set_ctx_md_params_fn oqs_sig_set_ctx_md_params;
 static OSSL_FUNC_signature_settable_ctx_md_params_fn oqs_sig_settable_ctx_md_params;
 
 // OIDS:
 static int get_oqs_oid(unsigned char* oidbuf, const char *oqs_name) {
 ///// OQS_TEMPLATE_FRAGMENT_SIG_OIDS_START
-   if (!strcmp(OQS_SIG_alg_default, oqs_name))
+   if (!strcmp("DEFAULT", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.1.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_2, oqs_name))
+   if (!strcmp("DILITHIUM_2", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.7.4.4", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_3, oqs_name))
+   if (!strcmp("DILITHIUM_3", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.7.6.5", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_5, oqs_name))
+   if (!strcmp("DILITHIUM_5", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.7.8.7", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_2_aes, oqs_name))
+   if (!strcmp("DILITHIUM_2_AES", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.11.4.4", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_3_aes, oqs_name))
+   if (!strcmp("DILITHIUM_3_AES", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.11.6.5", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_dilithium_5_aes, oqs_name))
+   if (!strcmp("DILITHIUM_5_AES", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.2.267.11.8.7", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_falcon_512, oqs_name))
+   if (!strcmp("FALCON_512", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.3.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_falcon_1024, oqs_name))
+   if (!strcmp("FALCON_1024", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.3.4", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_picnic_L1_full, oqs_name))
+   if (!strcmp("PICNIC_L1_FULL", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.311.89.2.1.7", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_picnic3_L1, oqs_name))
+   if (!strcmp("PICNIC3_L1", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.6.1.4.1.311.89.2.1.21", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_rainbow_I_classic, oqs_name))
+   if (!strcmp("RAINBOW_I_CLASSIC", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.5.1.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_rainbow_V_classic, oqs_name))
+   if (!strcmp("RAINBOW_V_CLASSIC", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.5.3.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_sphincs_haraka_128f_robust, oqs_name))
+   if (!strcmp("SPHINCS_HARAKA_128F_ROBUST", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.6.1.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_sphincs_sha256_128f_robust, oqs_name))
+   if (!strcmp("SPHINCS_SHA256_128F_ROBUST", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.6.4.1", 1), &oidbuf);
    else
-   if (!strcmp(OQS_SIG_alg_sphincs_shake256_128f_robust, oqs_name))
+   if (!strcmp("SPHINCS_SHAKE256_128F_ROBUST", oqs_name))
        return i2d_ASN1_OBJECT(OBJ_txt2obj("1.3.9999.6.7.1", 1), &oidbuf);
    else
 ///// OQS_TEMPLATE_FRAGMENT_SIG_OIDS_END
