diff --git a/oqsprov/oqsdecoders.inc b/oqsprov/oqsdecoders.inc
index cfa6c06..f7e1141 100644
--- a/oqsprov/oqsdecoders.inc
+++ b/oqsprov/oqsdecoders.inc
@@ -72,6 +72,14 @@ DECODER_w_structure("frodo1344shake", der, SubjectPublicKeyInfo, frodo1344shake)
 DECODER_w_structure("p521_frodo1344shake", der, PrivateKeyInfo, p521_frodo1344shake),
 DECODER_w_structure("p521_frodo1344shake", der, SubjectPublicKeyInfo, p521_frodo1344shake),
 #endif
+#ifdef OQS_ENABLE_KEM_kyber_1024
+DECODER_w_structure("kyber1024", der, PrivateKeyInfo, kyber1024),
+DECODER_w_structure("kyber1024", der, SubjectPublicKeyInfo, kyber1024),
+DECODER_w_structure("p521_kyber1024", der, PrivateKeyInfo, p521_kyber1024),
+DECODER_w_structure("p521_kyber1024", der, SubjectPublicKeyInfo, p521_kyber1024),
+DECODER_w_structure("secp521_r1_kyber1024", der, PrivateKeyInfo, secp521_r1_kyber1024),
+DECODER_w_structure("secp521_r1_kyber1024", der, SubjectPublicKeyInfo, secp521_r1_kyber1024),
+#endif
 #ifdef OQS_ENABLE_KEM_ml_kem_512
 DECODER_w_structure("mlkem512", der, PrivateKeyInfo, mlkem512),
 DECODER_w_structure("mlkem512", der, SubjectPublicKeyInfo, mlkem512),
