diff --git a/oqsprov/oqsencoders.inc b/oqsprov/oqsencoders.inc
index 53ca748..ef0be6b 100644
--- a/oqsprov/oqsencoders.inc
+++ b/oqsprov/oqsencoders.inc
@@ -200,6 +200,29 @@ ENCODER_w_structure("p521_frodo1344shake", p521_frodo1344shake, der, SubjectPubl
 ENCODER_w_structure("p521_frodo1344shake", p521_frodo1344shake, pem, SubjectPublicKeyInfo),
 ENCODER_TEXT("p521_frodo1344shake", p521_frodo1344shake),
 #endif
+#ifdef OQS_ENABLE_KEM_kyber_1024
+ENCODER_w_structure("kyber1024", kyber1024, der, PrivateKeyInfo),
+ENCODER_w_structure("kyber1024", kyber1024, pem, PrivateKeyInfo),
+ENCODER_w_structure("kyber1024", kyber1024, der, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("kyber1024", kyber1024, pem, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("kyber1024", kyber1024, der, SubjectPublicKeyInfo),
+ENCODER_w_structure("kyber1024", kyber1024, pem, SubjectPublicKeyInfo),
+ENCODER_TEXT("kyber1024", kyber1024),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, der, PrivateKeyInfo),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, pem, PrivateKeyInfo),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, der, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, pem, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, der, SubjectPublicKeyInfo),
+ENCODER_w_structure("p521_kyber1024", p521_kyber1024, pem, SubjectPublicKeyInfo),
+ENCODER_TEXT("p521_kyber1024", p521_kyber1024),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, der, PrivateKeyInfo),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, pem, PrivateKeyInfo),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, der, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, pem, EncryptedPrivateKeyInfo),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, der, SubjectPublicKeyInfo),
+ENCODER_w_structure("secp521_r1_kyber1024", secp521_r1_kyber1024, pem, SubjectPublicKeyInfo),
+ENCODER_TEXT("secp521_r1_kyber1024", secp521_r1_kyber1024),
+#endif
 #ifdef OQS_ENABLE_KEM_ml_kem_512
 ENCODER_w_structure("mlkem512", mlkem512, der, PrivateKeyInfo),
 ENCODER_w_structure("mlkem512", mlkem512, pem, PrivateKeyInfo),
