diff --git a/oqsprov/oqsprov_keys.c b/oqsprov/oqsprov_keys.c
index 4cf664d..da42478 100644
--- a/oqsprov/oqsprov_keys.c
+++ b/oqsprov/oqsprov_keys.c
@@ -41,7 +41,13 @@ typedef enum { KEY_OP_PUBLIC, KEY_OP_PRIVATE, KEY_OP_KEYGEN } oqsx_key_op_t;
 
 typedef struct {
     int nid;
+#ifdef __MVS__
+    __attribute__((encoding("ascii")))
+#endif
     char *tlsname;
+#ifdef __MVS__
+    __attribute__((encoding("ascii")))
+#endif
     char *oqsname;
     int keytype;
     int secbits;
@@ -52,8 +58,12 @@ static int oqsx_key_recreate_classickey(OQSX_KEY *key, oqsx_key_op_t op);
 
 ///// OQS_TEMPLATE_FRAGMENT_OQSNAMES_START
 
+
+
+
+
 #ifdef OQS_KEM_ENCODERS
-#define NID_TABLE_LEN 105
+#define NID_TABLE_LEN 108
 #else
 #define NID_TABLE_LEN 67
 #endif
@@ -61,166 +71,117 @@ static int oqsx_key_recreate_classickey(OQSX_KEY *key, oqsx_key_op_t op);
 static oqs_nid_name_t nid_names[NID_TABLE_LEN] = {
 #ifdef OQS_KEM_ENCODERS
 
-    {0, "frodo640aes", OQS_KEM_alg_frodokem_640_aes, KEY_TYPE_KEM, 128, 0},
-    {0, "p256_frodo640aes", OQS_KEM_alg_frodokem_640_aes, KEY_TYPE_ECP_HYB_KEM,
-     128, 0},
-    {0, "x25519_frodo640aes", OQS_KEM_alg_frodokem_640_aes,
-     KEY_TYPE_ECX_HYB_KEM, 128, 0},
-    {0, "frodo640shake", OQS_KEM_alg_frodokem_640_shake, KEY_TYPE_KEM, 128, 0},
-    {0, "p256_frodo640shake", OQS_KEM_alg_frodokem_640_shake,
-     KEY_TYPE_ECP_HYB_KEM, 128, 0},
-    {0, "x25519_frodo640shake", OQS_KEM_alg_frodokem_640_shake,
-     KEY_TYPE_ECX_HYB_KEM, 128, 0},
-    {0, "frodo976aes", OQS_KEM_alg_frodokem_976_aes, KEY_TYPE_KEM, 192, 0},
-    {0, "p384_frodo976aes", OQS_KEM_alg_frodokem_976_aes, KEY_TYPE_ECP_HYB_KEM,
-     192, 0},
-    {0, "x448_frodo976aes", OQS_KEM_alg_frodokem_976_aes, KEY_TYPE_ECX_HYB_KEM,
-     192, 0},
-    {0, "frodo976shake", OQS_KEM_alg_frodokem_976_shake, KEY_TYPE_KEM, 192, 0},
-    {0, "p384_frodo976shake", OQS_KEM_alg_frodokem_976_shake,
-     KEY_TYPE_ECP_HYB_KEM, 192, 0},
-    {0, "x448_frodo976shake", OQS_KEM_alg_frodokem_976_shake,
-     KEY_TYPE_ECX_HYB_KEM, 192, 0},
-    {0, "frodo1344aes", OQS_KEM_alg_frodokem_1344_aes, KEY_TYPE_KEM, 256, 0},
-    {0, "p521_frodo1344aes", OQS_KEM_alg_frodokem_1344_aes,
-     KEY_TYPE_ECP_HYB_KEM, 256, 0},
-    {0, "frodo1344shake", OQS_KEM_alg_frodokem_1344_shake, KEY_TYPE_KEM, 256,
-     0},
-    {0, "p521_frodo1344shake", OQS_KEM_alg_frodokem_1344_shake,
-     KEY_TYPE_ECP_HYB_KEM, 256, 0},
-    {0, "mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_KEM, 128, 0},
-    {0, "p256_mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_ECP_HYB_KEM, 128, 0},
-    {0, "x25519_mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_ECX_HYB_KEM, 128,
-     1},
-    {0, "bp256_mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_ECBP_HYB_KEM, 128,
-     1},
-    {0, "mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_KEM, 192, 0},
-    {0, "p384_mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECP_HYB_KEM, 192, 0},
-    {0, "x448_mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECX_HYB_KEM, 192, 1},
-    {0, "bp384_mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECBP_HYB_KEM, 192,
-     1},
-    {0, "X25519MLKEM768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECX_HYB_KEM, 192, 1},
-    {0, "SecP256r1MLKEM768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECP_HYB_KEM, 192,
-     0},
-    {0, "mlkem1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_KEM, 256, 0},
-    {0, "p521_mlkem1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_ECP_HYB_KEM, 256,
-     0},
-    {0, "SecP384r1MLKEM1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_ECP_HYB_KEM,
-     256, 0},
-    {0, "bp512_mlkem1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_ECBP_HYB_KEM, 256,
-     1},
-    {0, "bikel1", OQS_KEM_alg_bike_l1, KEY_TYPE_KEM, 128, 0},
-    {0, "p256_bikel1", OQS_KEM_alg_bike_l1, KEY_TYPE_ECP_HYB_KEM, 128, 0},
-    {0, "x25519_bikel1", OQS_KEM_alg_bike_l1, KEY_TYPE_ECX_HYB_KEM, 128, 0},
-    {0, "bikel3", OQS_KEM_alg_bike_l3, KEY_TYPE_KEM, 192, 0},
-    {0, "p384_bikel3", OQS_KEM_alg_bike_l3, KEY_TYPE_ECP_HYB_KEM, 192, 0},
-    {0, "x448_bikel3", OQS_KEM_alg_bike_l3, KEY_TYPE_ECX_HYB_KEM, 192, 0},
-    {0, "bikel5", OQS_KEM_alg_bike_l5, KEY_TYPE_KEM, 256, 0},
-    {0, "p521_bikel5", OQS_KEM_alg_bike_l5, KEY_TYPE_ECP_HYB_KEM, 256, 0},
+  { 0, "frodo640aes", OQS_KEM_alg_frodokem_640_aes, KEY_TYPE_KEM, 128, 0 },
+  { 0, "p256_frodo640aes", OQS_KEM_alg_frodokem_640_aes, KEY_TYPE_ECP_HYB_KEM , 128, 0 },
+  { 0, "x25519_frodo640aes", OQS_KEM_alg_frodokem_640_aes,   KEY_TYPE_ECX_HYB_KEM  , 128, 0 },
+  { 0, "frodo640shake", OQS_KEM_alg_frodokem_640_shake, KEY_TYPE_KEM, 128, 0 },
+  { 0, "p256_frodo640shake", OQS_KEM_alg_frodokem_640_shake, KEY_TYPE_ECP_HYB_KEM , 128, 0 },
+  { 0, "x25519_frodo640shake", OQS_KEM_alg_frodokem_640_shake,   KEY_TYPE_ECX_HYB_KEM  , 128, 0 },
+  { 0, "frodo976aes", OQS_KEM_alg_frodokem_976_aes, KEY_TYPE_KEM, 192, 0 },
+  { 0, "p384_frodo976aes", OQS_KEM_alg_frodokem_976_aes, KEY_TYPE_ECP_HYB_KEM , 192, 0 },
+  { 0, "x448_frodo976aes", OQS_KEM_alg_frodokem_976_aes,   KEY_TYPE_ECX_HYB_KEM  , 192, 0 },
+  { 0, "frodo976shake", OQS_KEM_alg_frodokem_976_shake, KEY_TYPE_KEM, 192, 0 },
+  { 0, "p384_frodo976shake", OQS_KEM_alg_frodokem_976_shake, KEY_TYPE_ECP_HYB_KEM , 192, 0 },
+  { 0, "x448_frodo976shake", OQS_KEM_alg_frodokem_976_shake,   KEY_TYPE_ECX_HYB_KEM  , 192, 0 },
+  { 0, "frodo1344aes", OQS_KEM_alg_frodokem_1344_aes, KEY_TYPE_KEM, 256, 0 },
+  { 0, "p521_frodo1344aes", OQS_KEM_alg_frodokem_1344_aes, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "frodo1344shake", OQS_KEM_alg_frodokem_1344_shake, KEY_TYPE_KEM, 256, 0 },
+  { 0, "p521_frodo1344shake", OQS_KEM_alg_frodokem_1344_shake, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "kyber1024", OQS_KEM_alg_kyber_1024, KEY_TYPE_KEM, 256, 0 },
+  { 0, "p521_kyber1024", OQS_KEM_alg_kyber_1024, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "secp521_r1_kyber1024", OQS_KEM_alg_kyber_1024, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_KEM, 128, 0 },
+  { 0, "p256_mlkem512", OQS_KEM_alg_ml_kem_512, KEY_TYPE_ECP_HYB_KEM , 128, 0 },
+  { 0, "x25519_mlkem512", OQS_KEM_alg_ml_kem_512,   KEY_TYPE_ECX_HYB_KEM  , 128, 1 },
+  { 0, "bp256_mlkem512", OQS_KEM_alg_ml_kem_512,  KEY_TYPE_ECBP_HYB_KEM  , 128, 1 },
+  { 0, "mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_KEM, 192, 0 },
+  { 0, "p384_mlkem768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECP_HYB_KEM , 192, 0 },
+  { 0, "x448_mlkem768", OQS_KEM_alg_ml_kem_768,   KEY_TYPE_ECX_HYB_KEM  , 192, 1 },
+  { 0, "bp384_mlkem768", OQS_KEM_alg_ml_kem_768,  KEY_TYPE_ECBP_HYB_KEM  , 192, 1 },
+  { 0, "X25519MLKEM768", OQS_KEM_alg_ml_kem_768,   KEY_TYPE_ECX_HYB_KEM  , 192, 1 },
+  { 0, "SecP256r1MLKEM768", OQS_KEM_alg_ml_kem_768, KEY_TYPE_ECP_HYB_KEM , 192, 0 },
+  { 0, "mlkem1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_KEM, 256, 0 },
+  { 0, "p521_mlkem1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "SecP384r1MLKEM1024", OQS_KEM_alg_ml_kem_1024, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
+  { 0, "bp512_mlkem1024", OQS_KEM_alg_ml_kem_1024,  KEY_TYPE_ECBP_HYB_KEM  , 256, 1 },
+  { 0, "bikel1", OQS_KEM_alg_bike_l1, KEY_TYPE_KEM, 128, 0 },
+  { 0, "p256_bikel1", OQS_KEM_alg_bike_l1, KEY_TYPE_ECP_HYB_KEM , 128, 0 },
+  { 0, "x25519_bikel1", OQS_KEM_alg_bike_l1,   KEY_TYPE_ECX_HYB_KEM  , 128, 0 },
+  { 0, "bikel3", OQS_KEM_alg_bike_l3, KEY_TYPE_KEM, 192, 0 },
+  { 0, "p384_bikel3", OQS_KEM_alg_bike_l3, KEY_TYPE_ECP_HYB_KEM , 192, 0 },
+  { 0, "x448_bikel3", OQS_KEM_alg_bike_l3,   KEY_TYPE_ECX_HYB_KEM  , 192, 0 },
+  { 0, "bikel5", OQS_KEM_alg_bike_l5, KEY_TYPE_KEM, 256, 0 },
+  { 0, "p521_bikel5", OQS_KEM_alg_bike_l5, KEY_TYPE_ECP_HYB_KEM , 256, 0 },
 
 #endif /* OQS_KEM_ENCODERS */
-    {0, "mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_SIG, 128},
-    {0, "p256_mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_HYB_SIG, 128},
-    {0, "rsa3072_mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_HYB_SIG, 128},
-    {0, "mldsa65", OQS_SIG_alg_ml_dsa_65, KEY_TYPE_SIG, 192},
-    {0, "p384_mldsa65", OQS_SIG_alg_ml_dsa_65, KEY_TYPE_HYB_SIG, 192},
-    {0, "mldsa87", OQS_SIG_alg_ml_dsa_87, KEY_TYPE_SIG, 256},
-    {0, "p521_mldsa87", OQS_SIG_alg_ml_dsa_87, KEY_TYPE_HYB_SIG, 256},
-    {0, "falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_SIG, 128},
-    {0, "p256_falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_HYB_SIG, 128},
-    {0, "rsa3072_falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_HYB_SIG, 128},
-    {0, "falconpadded512", OQS_SIG_alg_falcon_padded_512, KEY_TYPE_SIG, 128},
-    {0, "p256_falconpadded512", OQS_SIG_alg_falcon_padded_512, KEY_TYPE_HYB_SIG,
-     128},
-    {0, "rsa3072_falconpadded512", OQS_SIG_alg_falcon_padded_512,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "falcon1024", OQS_SIG_alg_falcon_1024, KEY_TYPE_SIG, 256},
-    {0, "p521_falcon1024", OQS_SIG_alg_falcon_1024, KEY_TYPE_HYB_SIG, 256},
-    {0, "falconpadded1024", OQS_SIG_alg_falcon_padded_1024, KEY_TYPE_SIG, 256},
-    {0, "p521_falconpadded1024", OQS_SIG_alg_falcon_padded_1024,
-     KEY_TYPE_HYB_SIG, 256},
-    {0, "sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple,
-     KEY_TYPE_SIG, 128},
-    {0, "p256_sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "rsa3072_sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple,
-     KEY_TYPE_SIG, 128},
-    {0, "p256_sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "rsa3072_sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "sphincssha2192fsimple", OQS_SIG_alg_sphincs_sha2_192f_simple,
-     KEY_TYPE_SIG, 192},
-    {0, "p384_sphincssha2192fsimple", OQS_SIG_alg_sphincs_sha2_192f_simple,
-     KEY_TYPE_HYB_SIG, 192},
-    {0, "sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple,
-     KEY_TYPE_SIG, 128},
-    {0, "p256_sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "rsa3072_sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "mayo1", OQS_SIG_alg_mayo_1, KEY_TYPE_SIG, 128},
-    {0, "p256_mayo1", OQS_SIG_alg_mayo_1, KEY_TYPE_HYB_SIG, 128},
-    {0, "mayo2", OQS_SIG_alg_mayo_2, KEY_TYPE_SIG, 128},
-    {0, "p256_mayo2", OQS_SIG_alg_mayo_2, KEY_TYPE_HYB_SIG, 128},
-    {0, "mayo3", OQS_SIG_alg_mayo_3, KEY_TYPE_SIG, 192},
-    {0, "p384_mayo3", OQS_SIG_alg_mayo_3, KEY_TYPE_HYB_SIG, 192},
-    {0, "mayo5", OQS_SIG_alg_mayo_5, KEY_TYPE_SIG, 256},
-    {0, "p521_mayo5", OQS_SIG_alg_mayo_5, KEY_TYPE_HYB_SIG, 256},
-    {0, "CROSSrsdp128balanced", OQS_SIG_alg_cross_rsdp_128_balanced,
-     KEY_TYPE_SIG, 128},
-    {0, "OV_Is_pkc", OQS_SIG_alg_uov_ov_Is_pkc, KEY_TYPE_SIG, 128},
-    {0, "p256_OV_Is_pkc", OQS_SIG_alg_uov_ov_Is_pkc, KEY_TYPE_HYB_SIG, 128},
-    {0, "OV_Ip_pkc", OQS_SIG_alg_uov_ov_Ip_pkc, KEY_TYPE_SIG, 128},
-    {0, "p256_OV_Ip_pkc", OQS_SIG_alg_uov_ov_Ip_pkc, KEY_TYPE_HYB_SIG, 128},
-    {0, "OV_Is_pkc_skc", OQS_SIG_alg_uov_ov_Is_pkc_skc, KEY_TYPE_SIG, 128},
-    {0, "p256_OV_Is_pkc_skc", OQS_SIG_alg_uov_ov_Is_pkc_skc, KEY_TYPE_HYB_SIG,
-     128},
-    {0, "OV_Ip_pkc_skc", OQS_SIG_alg_uov_ov_Ip_pkc_skc, KEY_TYPE_SIG, 128},
-    {0, "p256_OV_Ip_pkc_skc", OQS_SIG_alg_uov_ov_Ip_pkc_skc, KEY_TYPE_HYB_SIG,
-     128},
-    {0, "snova2454", OQS_SIG_alg_snova_SNOVA_24_5_4, KEY_TYPE_SIG, 128},
-    {0, "p256_snova2454", OQS_SIG_alg_snova_SNOVA_24_5_4, KEY_TYPE_HYB_SIG,
-     128},
-    {0, "snova2454esk", OQS_SIG_alg_snova_SNOVA_24_5_4_esk, KEY_TYPE_SIG, 128},
-    {0, "p256_snova2454esk", OQS_SIG_alg_snova_SNOVA_24_5_4_esk,
-     KEY_TYPE_HYB_SIG, 128},
-    {0, "snova37172", OQS_SIG_alg_snova_SNOVA_37_17_2, KEY_TYPE_SIG, 128},
-    {0, "p256_snova37172", OQS_SIG_alg_snova_SNOVA_37_17_2, KEY_TYPE_HYB_SIG,
-     128},
-    {0, "snova2455", OQS_SIG_alg_snova_SNOVA_24_5_5, KEY_TYPE_SIG, 192},
-    {0, "p384_snova2455", OQS_SIG_alg_snova_SNOVA_24_5_5, KEY_TYPE_HYB_SIG,
-     192},
-    {0, "snova2965", OQS_SIG_alg_snova_SNOVA_29_6_5, KEY_TYPE_SIG, 256},
-    {0, "p521_snova2965", OQS_SIG_alg_snova_SNOVA_29_6_5, KEY_TYPE_HYB_SIG,
-     256},
-    {0, "slhdsasha2128s", OQS_SIG_alg_slh_dsa_pure_sha2_128s, KEY_TYPE_SIG,
-     128},
-    {0, "slhdsasha2128f", OQS_SIG_alg_slh_dsa_pure_sha2_128f, KEY_TYPE_SIG,
-     128},
-    {0, "slhdsasha2192s", OQS_SIG_alg_slh_dsa_pure_sha2_192s, KEY_TYPE_SIG,
-     192},
-    {0, "slhdsasha2192f", OQS_SIG_alg_slh_dsa_pure_sha2_192f, KEY_TYPE_SIG,
-     192},
-    {0, "slhdsasha2256s", OQS_SIG_alg_slh_dsa_pure_sha2_256s, KEY_TYPE_SIG,
-     256},
-    {0, "slhdsasha2256f", OQS_SIG_alg_slh_dsa_pure_sha2_256f, KEY_TYPE_SIG,
-     256},
-    {0, "slhdsashake128s", OQS_SIG_alg_slh_dsa_pure_shake_128s, KEY_TYPE_SIG,
-     128},
-    {0, "slhdsashake128f", OQS_SIG_alg_slh_dsa_pure_shake_128f, KEY_TYPE_SIG,
-     128},
-    {0, "slhdsashake192s", OQS_SIG_alg_slh_dsa_pure_shake_192s, KEY_TYPE_SIG,
-     192},
-    {0, "slhdsashake192f", OQS_SIG_alg_slh_dsa_pure_shake_192f, KEY_TYPE_SIG,
-     192},
-    {0, "slhdsashake256s", OQS_SIG_alg_slh_dsa_pure_shake_256s, KEY_TYPE_SIG,
-     256},
-    {0, "slhdsashake256f", OQS_SIG_alg_slh_dsa_pure_shake_256f, KEY_TYPE_SIG,
-     256},
-    ///// OQS_TEMPLATE_FRAGMENT_OQSNAMES_END
+       { 0, "mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_SIG, 128 },
+       { 0, "p256_mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_mldsa44", OQS_SIG_alg_ml_dsa_44, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "mldsa65", OQS_SIG_alg_ml_dsa_65, KEY_TYPE_SIG, 192 },
+       { 0, "p384_mldsa65", OQS_SIG_alg_ml_dsa_65, KEY_TYPE_HYB_SIG, 192 },
+       { 0, "mldsa87", OQS_SIG_alg_ml_dsa_87, KEY_TYPE_SIG, 256 },
+       { 0, "p521_mldsa87", OQS_SIG_alg_ml_dsa_87, KEY_TYPE_HYB_SIG, 256 },
+       { 0, "falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_SIG, 128 },
+       { 0, "p256_falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_falcon512", OQS_SIG_alg_falcon_512, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "falconpadded512", OQS_SIG_alg_falcon_padded_512, KEY_TYPE_SIG, 128 },
+       { 0, "p256_falconpadded512", OQS_SIG_alg_falcon_padded_512, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_falconpadded512", OQS_SIG_alg_falcon_padded_512, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "falcon1024", OQS_SIG_alg_falcon_1024, KEY_TYPE_SIG, 256 },
+       { 0, "p521_falcon1024", OQS_SIG_alg_falcon_1024, KEY_TYPE_HYB_SIG, 256 },
+       { 0, "falconpadded1024", OQS_SIG_alg_falcon_padded_1024, KEY_TYPE_SIG, 256 },
+       { 0, "p521_falconpadded1024", OQS_SIG_alg_falcon_padded_1024, KEY_TYPE_HYB_SIG, 256 },
+       { 0, "sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple, KEY_TYPE_SIG, 128 },
+       { 0, "p256_sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_sphincssha2128fsimple", OQS_SIG_alg_sphincs_sha2_128f_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple, KEY_TYPE_SIG, 128 },
+       { 0, "p256_sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_sphincssha2128ssimple", OQS_SIG_alg_sphincs_sha2_128s_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "sphincssha2192fsimple", OQS_SIG_alg_sphincs_sha2_192f_simple, KEY_TYPE_SIG, 192 },
+       { 0, "p384_sphincssha2192fsimple", OQS_SIG_alg_sphincs_sha2_192f_simple, KEY_TYPE_HYB_SIG, 192 },
+       { 0, "sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple, KEY_TYPE_SIG, 128 },
+       { 0, "p256_sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "rsa3072_sphincsshake128fsimple", OQS_SIG_alg_sphincs_shake_128f_simple, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "mayo1", OQS_SIG_alg_mayo_1, KEY_TYPE_SIG, 128 },
+       { 0, "p256_mayo1", OQS_SIG_alg_mayo_1, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "mayo2", OQS_SIG_alg_mayo_2, KEY_TYPE_SIG, 128 },
+       { 0, "p256_mayo2", OQS_SIG_alg_mayo_2, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "mayo3", OQS_SIG_alg_mayo_3, KEY_TYPE_SIG, 192 },
+       { 0, "p384_mayo3", OQS_SIG_alg_mayo_3, KEY_TYPE_HYB_SIG, 192 },
+       { 0, "mayo5", OQS_SIG_alg_mayo_5, KEY_TYPE_SIG, 256 },
+       { 0, "p521_mayo5", OQS_SIG_alg_mayo_5, KEY_TYPE_HYB_SIG, 256 },
+       { 0, "CROSSrsdp128balanced", OQS_SIG_alg_cross_rsdp_128_balanced, KEY_TYPE_SIG, 128 },
+       { 0, "OV_Is_pkc", OQS_SIG_alg_uov_ov_Is_pkc, KEY_TYPE_SIG, 128 },
+       { 0, "p256_OV_Is_pkc", OQS_SIG_alg_uov_ov_Is_pkc, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "OV_Ip_pkc", OQS_SIG_alg_uov_ov_Ip_pkc, KEY_TYPE_SIG, 128 },
+       { 0, "p256_OV_Ip_pkc", OQS_SIG_alg_uov_ov_Ip_pkc, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "OV_Is_pkc_skc", OQS_SIG_alg_uov_ov_Is_pkc_skc, KEY_TYPE_SIG, 128 },
+       { 0, "p256_OV_Is_pkc_skc", OQS_SIG_alg_uov_ov_Is_pkc_skc, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "OV_Ip_pkc_skc", OQS_SIG_alg_uov_ov_Ip_pkc_skc, KEY_TYPE_SIG, 128 },
+       { 0, "p256_OV_Ip_pkc_skc", OQS_SIG_alg_uov_ov_Ip_pkc_skc, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "snova2454", OQS_SIG_alg_snova_SNOVA_24_5_4, KEY_TYPE_SIG, 128 },
+       { 0, "p256_snova2454", OQS_SIG_alg_snova_SNOVA_24_5_4, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "snova2454esk", OQS_SIG_alg_snova_SNOVA_24_5_4_esk, KEY_TYPE_SIG, 128 },
+       { 0, "p256_snova2454esk", OQS_SIG_alg_snova_SNOVA_24_5_4_esk, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "snova37172", OQS_SIG_alg_snova_SNOVA_37_17_2, KEY_TYPE_SIG, 128 },
+       { 0, "p256_snova37172", OQS_SIG_alg_snova_SNOVA_37_17_2, KEY_TYPE_HYB_SIG, 128 },
+       { 0, "snova2455", OQS_SIG_alg_snova_SNOVA_24_5_5, KEY_TYPE_SIG, 192 },
+       { 0, "p384_snova2455", OQS_SIG_alg_snova_SNOVA_24_5_5, KEY_TYPE_HYB_SIG, 192 },
+       { 0, "snova2965", OQS_SIG_alg_snova_SNOVA_29_6_5, KEY_TYPE_SIG, 256 },
+       { 0, "p521_snova2965", OQS_SIG_alg_snova_SNOVA_29_6_5, KEY_TYPE_HYB_SIG, 256 },
+       { 0, "slhdsasha2128s", OQS_SIG_alg_slh_dsa_pure_sha2_128s, KEY_TYPE_SIG, 128 },
+       { 0, "slhdsasha2128f", OQS_SIG_alg_slh_dsa_pure_sha2_128f, KEY_TYPE_SIG, 128 },
+       { 0, "slhdsasha2192s", OQS_SIG_alg_slh_dsa_pure_sha2_192s, KEY_TYPE_SIG, 192 },
+       { 0, "slhdsasha2192f", OQS_SIG_alg_slh_dsa_pure_sha2_192f, KEY_TYPE_SIG, 192 },
+       { 0, "slhdsasha2256s", OQS_SIG_alg_slh_dsa_pure_sha2_256s, KEY_TYPE_SIG, 256 },
+       { 0, "slhdsasha2256f", OQS_SIG_alg_slh_dsa_pure_sha2_256f, KEY_TYPE_SIG, 256 },
+       { 0, "slhdsashake128s", OQS_SIG_alg_slh_dsa_pure_shake_128s, KEY_TYPE_SIG, 128 },
+       { 0, "slhdsashake128f", OQS_SIG_alg_slh_dsa_pure_shake_128f, KEY_TYPE_SIG, 128 },
+       { 0, "slhdsashake192s", OQS_SIG_alg_slh_dsa_pure_shake_192s, KEY_TYPE_SIG, 192 },
+       { 0, "slhdsashake192f", OQS_SIG_alg_slh_dsa_pure_shake_192f, KEY_TYPE_SIG, 192 },
+       { 0, "slhdsashake256s", OQS_SIG_alg_slh_dsa_pure_shake_256s, KEY_TYPE_SIG, 256 },
+       { 0, "slhdsashake256f", OQS_SIG_alg_slh_dsa_pure_shake_256f, KEY_TYPE_SIG, 256 },
+///// OQS_TEMPLATE_FRAGMENT_OQSNAMES_END
 };
 
 int oqs_set_nid(char *tlsname, int nid) {
@@ -716,15 +677,28 @@ static OQSX_KEY *oqsx_key_op(const X509_ALGOR *palg, const unsigned char *p,
 
     OQS_KEY_PRINTF2("OQSX KEY: key_op called with data of len %d\n", plen);
     if (palg != NULL) {
-        int ptype;
-
-        /* Algorithm parameters must be absent */
-        X509_ALGOR_get0(NULL, &ptype, NULL, palg);
-        if (ptype != V_ASN1_UNDEF || !palg || !palg->algorithm) {
-            ERR_raise(ERR_LIB_USER, OQSPROV_R_INVALID_ENCODING);
-            return 0;
-        }
+        // First get the nid from the algorithm OID
         nid = OBJ_obj2nid(palg->algorithm);
+        // Then get the keytype for the determined nid
+        int current_keytype = get_keytype(nid);
+
+        /* Algorithm parameters must be absent for non-ECP/ECBP hybrid KEMs.
+         * For ECP/ECBP hybrid KEMs, parameters (curve OID) must be present.
+         */
+        if (current_keytype == KEY_TYPE_ECP_HYB_KEM ||
+            current_keytype == KEY_TYPE_ECBP_HYB_KEM) {
+            // Parameters must be present and of type V_ASN1_OBJECT for ECP/ECBP hybrid KEMs
+            if (palg->parameter == NULL || OBJ_obj2nid(palg->parameter->value.object) == NID_undef) {
+                ERR_raise(ERR_LIB_USER, OQSPROV_R_INVALID_ENCODING);
+                return 0;
+            }
+        } else {
+            // For other key types, parameters must be absent
+            if (palg->parameter != NULL) {
+                ERR_raise(ERR_LIB_USER, OQSPROV_R_INVALID_ENCODING);
+                return 0;
+            }
+        }
     }
 
     if (p == NULL || nid == EVP_PKEY_NONE || nid == NID_undef) {
@@ -742,7 +716,7 @@ static OQSX_KEY *oqsx_key_op(const X509_ALGOR *palg, const unsigned char *p,
     if (op == KEY_OP_PUBLIC) {
         if (key->pubkeylen != plen) {
             ERR_raise(ERR_LIB_USER, OQSPROV_R_INVALID_ENCODING);
-            goto err_key_op;
+            return 0;
         }
         if (oqsx_key_allocate_keymaterial(key, 0)) {
             ERR_raise(ERR_LIB_USER, ERR_R_MALLOC_FAILURE);
